src/main/scala/edu/ie3/mobsim/utils/PathsAndSources.scala

```diff
@@ -199,17 +181,23 @@ object PathsAndSources extends LazyLogging {
     */
   private def determineRecentOutputDirectory(baseOutputDir: String): String = {
     /* find last modified directory in output path -> this is where SIMONA writes into during the simulation */
-    val dir = Paths
-      .get(baseOutputDir)
-      .toFile
+    val outputDir = new File(baseOutputDir)
+    val outputPathFiles = outputDir
       .listFiles(DIRECTORY.asInstanceOf[FileFilter])
-      .maxByOption(_.lastModified())
-      .getOrElse {
-        logger.warn(
-          "Unable to determine most recent output directory. Write to base path!"
-        )
-        baseOutputDir
-      }
+
+    val dir = if (!outputDir.exists() || outputPathFiles.isEmpty) {
```

Comment:
Ok got it now. It's all good, just one last suggestion: we don't need to check if outputPathFiles is empty, since `maxByOption` below already takes care of that.

```suggestion
    val dir = if (!outputDir.exists()) {
```